### PR TITLE
76379 registration complete

### PR DESCRIPTION
**Pull request by stevelee-civica**
add registration complete screen
add unit tests

### CODE REVIEW

File: src/EA.Weee.Web/Controllers/OrganisationRegistrationController.cs [hunk 1]
```diff
@@ -690,5 +690,18 @@ public async Task<ActionResult> ContactDetails(ContactDetailsViewModel model)
...
+
+        [HttpGet]
+        public ViewResult RegistrationComplete()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult RegistrationComplete(string searchTerm)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Controllers/OrganisationRegistrationController.cs:702]: @stevelee-civica We dont need the search term parameter I think.

File: src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs [hunk 2]
```diff
@@ -1455,5 +1455,32 @@ public async Task RepresentingCompanyRedirect_ShouldRedirectToType_WhenTransacti
             result.RouteValues["action"].Should().Be("Type");
             result.RouteValues["controller"].Should().Be("OrganisationRegistration");
         }
+
+        [Fact]
+        public void RegistrationComplete_Get_ReturnsView()
+        {
+            // Arrange
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs:1462]: @stevelee-civica  dont worry about an Arrange if there is nothing to arrange :)

**Comment by JohnPadgettCivica** [comment]
@stevelee-civica  looks like there is a merge conflict to resolve. I think this new page needs to be re-directed to after the Representing company details, Registered Company Details / Sole Trader / Partner details POST actions, like inside
image

It should go to the registration completed screen. Otherwise there is no current way to get to it?

File: src/EA.Weee.Web/Views/OrganisationRegistration/RegistrationComplete.cshtml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+@section full_main_content
+{
+    @using (Html.BeginForm())
+    {
+        @Html.AntiForgeryToken()
+        @Html.Gds().ValidationSummary()
+
+        <div class="govuk-panel govuk-panel--confirmation">
+            <h1 class="govuk-panel__title">
+                Organisation created
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/OrganisationRegistration/RegistrationComplete.cshtml:15]: @stevelee-civica  can use @title here

File: src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs (hunk 2 above)
Comment (stevelee-civica) [reply src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs]: removed

File: src/EA.Weee.Web/Views/OrganisationRegistration/RegistrationComplete.cshtml (hunk 3 above)
Comment (stevelee-civica) [reply src/EA.Weee.Web/Views/OrganisationRegistration/RegistrationComplete.cshtml:15]: updated

File: src/EA.Weee.Web/Controllers/OrganisationRegistrationController.cs [hunk 4]
```diff
@@ -302,12 +302,22 @@ public async Task<ActionResult> RepresentingCompanyDetails(RepresentingCompanyDe
...
+        {
+            if (model != null) 
+            { 
+                await transactionService.CaptureData(User.GetAccessToken(), model);
 
-            return RedirectToAction("Index", typeof(HoldingController).GetControllerName());
+                await transactionService.CompleteTransaction(User.GetAccessToken());
+                await cache.InvalidateOrganisationSearch();
+
+                return RegistrationComplete();
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Controllers/OrganisationRegistrationController.cs:317]: @stevelee-civica  Update this to return RedirectToAction(nameof(RegistrationComplete), typeof(OrganisationRegistrationController).GetControllerName());

File: src/EA.Weee.Web/Views/OrganisationRegistration/RegistrationComplete.cshtml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        @Html.Gds().ValidationSummary()
+
+        <div class="govuk-panel govuk-panel--confirmation">
+            <h1 class="govuk-panel__title">
+                @title
+            </h1>
+        </div>
+
+        <p>Your organisation has been created.</p>
+        <h3 class="govuk-heading-m">What happens next</h3>
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/OrganisationRegistration/RegistrationComplete.cshtml:20]: @stevelee-civica  Sorry missed this, change this to h2 as Wave will moan about heading heirachy

**Pull request merged**